Assets/Scripts/SpellTrigger.cs

```diff
@@ -1,17 +1,30 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
+using UnityEditorInternal;
 using UnityEngine;
 
 public class SpellTrigger : MonoBehaviour
 {
     public GameObject m_Projectile;    
-    public Transform m_SpawnTransform; 
+    public Transform m_SpawnTransform;
+    public IEnumerator coroutine;
 
-    private void Update()
+    private void FixedUpdate()
     {
         if (Input.GetKeyDown(KeyCode.Mouse0))
         {
-            Instantiate(m_Projectile, m_SpawnTransform.position, m_SpawnTransform.rotation);
+            StartCoroutine(Attack());
         }
     }
+
+    public IEnumerator Attack()
+    {    
+        Instantiate(m_Projectile, m_SpawnTransform.position, m_SpawnTransform.rotation);
+        yield return StartCoroutine(WaitForCooldown(3));
```

Comment:
Avoid using magic numbers for spell cooldown. Set a variable like float fireRate in the coroutine to avoid problems

Reply:
Instantiate(m_Projectile, m_SpawnTransform.position, m_SpawnTransform.rotation);
yield return StartCoroutine(WaitForCooldown(3));